Feature/gitignore

song1900 opened the pull request:
- .gitignore 파일을 추가
- .DS_Store, UserInterfaceState.xcuserstate 파일을 제거

song1900 committed:
Feat: gitignore 생성

song1900 committed:
Chore: .DS_Stroe 삭제

song1900 committed:
Chore: UserInterfaceState.xcuserstate 삭제

f-lab-jercy commented on GIL/GIL/.gitignore :
```diff
@@ -0,0 +1,62 @@
+# Created by https://www.toptal.com/developers/gitignore/api/macos,cocoapods,xcode
+# Edit at https://www.toptal.com/developers/gitignore?templates=macos,cocoapods,xcode
+
+### CocoaPods ###
+## CocoaPods GitIgnore Template
+
+# CocoaPods - Only use to conserve bandwidth / Save time on Pushing
+#           - Also handy if you have a large number of dependant pods
+#           - AS PER https://guides.cocoapods.org/using/using-cocoapods.html NEVER IGNORE THE LOCK FILE
```
의미 없는 주석들은 제거를 해도 좋을것 같습니다.

f-lab-jercy commented on GIL/GIL/.gitignore :
```diff
@@ -0,0 +1,62 @@
+## User settings
+xcuserdata/
+
+## Xcode 8 and earlier
+*.xcscmblueprint
+*.xccheckout
```
이 내용들은 왜 들어가 있을까요?

song1900 replied:
위 코드는 Xcode 8 이하 버전에서 생성되는 파일들을 제외하는 코드입니다.
지금 프로젝트에서는 필요 없는 내용으로 지우도록 하겠습니다.

f-lab-jercy commented on GIL/GIL/.gitignore :
```diff
@@ -0,0 +1,62 @@
+*.xccheckout
+
+### Xcode Patch ###
+*.xcodeproj/*
+!*.xcodeproj/project.pbxproj
+!*.xcodeproj/xcshareddata/
+!*.xcodeproj/project.xcworkspace/
+!*.xcworkspace/contents.xcworkspacedata
+/*.gcno
+**/xcshareddata/WorkspaceSettings.xcsettings
```
이 이그노어 값들의 각각의 역할이 어떻게 될까요?

song1900 replied:
제가 찾아본 결과 각각의 역할은 아래와 같습니다.

1. `*.xcodeproj/*` : 모든 '.xcodeproj' 디렉토리와 그 안의 파일을 무시하라는 의미입니다.

2. `!*.xcodeproj/project.pbxproj` : 'project.pbxproj' 파일을 무시 목록에서 제외하라는 의미입니다. 이 파일은 빌드 설정, 파일 참조 등 중요한 정보를 담고 있습니다.

3. `!*.xcodeproj/xcshareddata/` : 'xcshareddata' 디렉토리를 무시 목록에서 제외하라는 의미입니다. 이 디렉토리에는 프로젝트 설정이 저장되어 있습니다.

4. `!*.xcodeproj/project.xcworkspace/` : 'project.xcworkspace' 디렉토리를 무시 목록에서 제외하라는 의미입니다. 이 디렉토리에는 프로젝트의 워크스페이스 설정이 포함되어 있습니다.

5. `!*.xcworkspace/contents.xcworkspacedata` : '.xcworkspace' 디렉토리 내의 'contents.xcworkspacedata' 파일을 무시 목록에서 제외하라는 의미입니다. 이 파일에는 워크스페이스 구성 정보가 포함되어 있습니다.

6. `/*.gcno` : 루트 디렉토리에 위치한 '.gcno' 파일을 무시하라는 의미입니다. 이 파일은 코드 커버리지 분석을 위해 생성되며, 일반적으로 버전관리에 포함할 필요가 없다고 합니다.

7. `**/xcshareddata/WorkspaceSettings.xcsettings` : 모든 'xcshareddata' 디렉토리 내의 'WorkspaceSettings.xcsettings' 파일을 무시하라는 의미입니다. 이 파일은 로컬 작업 환경에 특화된 설정을 포함할 수 있고, 각각의 개발 환경에 따라 다를 수 있기 때문에 버전 관리에서 제외하는 것이 일반적이라고 합니다.

f-lab-jercy commented:
고생하셨습니다! 머지하셔도 좋습니다 코멘트만 한번 확인해주세요

song1900 committed:
Feat: .gitignore 수정

- Xcode 8 이하 에서 생성되는 파일을 제외해주는 ## Xcode 8 and earlier 제거 (현재 XCode 버전은 15.2)
- 의미 없는 주석 제거
